packages/grafana-data/src/transformations/transformers/join.ts

```diff
@@ -0,0 +1,93 @@
+import { DataFrame, DataTransformerInfo } from '../../types';
+import { DataTransformerID } from './ids';
+import { MutableDataFrame } from '../../dataframe';
+import { filterFieldsByNameTransformer } from './filterByName';
+import { ArrayVector } from '../../vector';
+
+export interface JoinOptions {
+  byField: string;
+}
+
+export const joinTransformer: DataTransformerInfo<JoinOptions> = {
+  id: DataTransformerID.join,
+  name: 'Join',
+  description: 'Append values into a single DataFrame.  This uses the name as the key',
+  defaultOptions: {},
+  transformer: options => (data: DataFrame[]) => {
+    const regex = `/^(${options.byField})$/`;
+    // not sure if I should use filterFieldsByNameTransformer to get the key field
+    const keyDataFrames = filterFieldsByNameTransformer.transformer({ include: regex })(data);
```

Comment:
I figured I'd use filterFieldsByNameTransformer here, but might not be optimal for performance?